src/iexamine.cpp

```diff
@@ -3474,6 +3477,51 @@ void iexamine::climb_down( player &p, const tripoint &examp )
     g->m.creature_on_trap( p );
 }
 
+player &player_on_couch( player &p, const tripoint autodoc_loc, player &null_patient, bool &adjacent_couch )
```

Comment:
Did you test moving out or destroying of autodoc couch? What's the point of having `adjacent_count` as a parameter? The code below doesn't check if it's true or false.

Reply:
adjacent_couch is assigned true during the search, and is checked on line 3539.

Reply:
I talked about `adjacent_couch` as a parameter of `player_on_couch` function. If `adjacent_couch` is a parameter, it should be checked inside the function.

Reply:
It is passed as false into the function, the function sets it true as a side effect of finding the couch. The point of having it as a paramter is to alter the value if the couch is adjacent to the autodoc.

Reply:
I think I got it. Though I think all this strange (for me) mechanics would not be needed if you haven't extracted `player_on_couch` function. I'm not a good programmer, but I intuitively feel that this is wrong, e.g.when function will do all it should do only when one of its parameters has a strictly specific name.

Reply:
I'm not 100% wild about it either, but:
patient is declared as a reference (player &patient), so it needs to be defined when it is declared.
patient is used throughout iexamine::autodoc, so I need to declare patient at the top scope of the function, and I can't declare it inside the if ( g->m.furn( couch_loc  ) == couch ) { scope.
I would rather not find the autodoc couch twice, so player_on_couch() should find the couch and the PC or NPC that is on it.

I could declare the return of player_on_couch() as std::pair<player &, bool> but that's honestly more hassle than it's worth comparing to just changing a parameter value as a side effect.